### PR TITLE
feat: Alloy type migration

### PR DESCRIPTION
**Pull request by DaniPopes**
First commit is 90% automated with regexes

TODO:
- [x] replace `B::random` with `random_with` whenever an Rng is available, https://github.com/alloy-rs/core/pull/282
- [x] ~~migrate `decode_revert_reason` to `alloy_sol_types`~~ another PR
- [ ] Revm migration: https://github.com/bluealloy/revm/pull/724
- [ ] Alloy release

### CODE REVIEW

File: crates/rpc/rpc/src/eth/api/sign.rs [hunk 1]
```diff
@@ -15,16 +15,16 @@ impl<Provider, Pool, Network> EthApi<Provider, Pool, Network> {
     pub(crate) async fn sign(&self, account: Address, message: Bytes) -> EthResult<Bytes> {
         let signer = self.find_signer(&account)?;
         let signature = signer.sign(account, &message).await?;
-        let bytes = hex::encode(signature.to_bytes()).as_bytes().into();
-        Ok(bytes)
+        let bytes = hex::encode(signature.to_bytes());
+        Ok(Bytes(bytes::Bytes::from(bytes)))
```
Comment (gakonst) [inline crates/rpc/rpc/src/eth/api/sign.rs:19]: maybe we can add a helper method on signature for this

File: crates/primitives/src/transaction/mod.rs [hunk 2]
```diff
@@ -836,10 +837,10 @@ impl TransactionSigned {
     /// Returns the enveloped encoded transactions.
     ///
     /// See also [TransactionSigned::encode_enveloped]
-    pub fn envelope_encoded(&self) -> bytes::Bytes {
+    pub fn envelope_encoded(&self) -> Bytes {
```
Comment (mattsse) [inline crates/primitives/src/transaction/mod.rs:840]: which bytes type is this here?

I'd like to keep bytes::Bytes and not a hex wrapper

**Review by mattsse — APPROVED** [review]
amazing, only 1 q re Bytes type

this PR is blocked by revm, right?

**Comment by DaniPopes** [comment]
Blocked by Revm migration, and Alloy release

File: crates/primitives/src/transaction/mod.rs (hunk 2 above)
Comment (DaniPopes) [reply crates/primitives/src/transaction/mod.rs:840]: This is alloy Bytes, I don't see why have multiple types in a public interface

**Comment by gakonst** [comment]
As discussed, let's no longer block this PR on the releases / merging of the deps, let's drive it to the end / remove ethers from as many places as we can, and merge on Monday after we release alpha.9

**Comment by DaniPopes** [comment]
Merging into `alloy` branch and continuing there

**Pull request merged**